### PR TITLE
SAMZA-1275: Kafka throws when users configure replication.factor for …

### PR DESCRIPTION
**Pull request by jmakes**
…Kafka default stream

### CODE REVIEW

**Comment by cpettitt-linkedin** [comment]
Code changes looks good. It would be great to have a test to verify filtered config is removed from the spec.

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java [hunk 1]
```diff
@@ -62,6 +63,26 @@ private static Properties mapToProperties(Map<String, String> map) {
   }
 
   /**
+   * Filter out properties from the original config that are not supported by Kafka.
+   * For example, we allow users to set replication.factor as a property of the streams
+   * and then parse it out so we can pass it separately as Kafka requires. But Kafka
+   * will also throw if replication.factor is passed as a property on a new topic.
```
Comment (navina) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java]: QQ: Is this behavior seen only when "creating" new topics?

Comment (jmakes) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java]: yes

**Comment by jmakes** [comment]
@cpettitt-linkedin Replying here since github won't let me directly

Unit test is coming, was just looking for feedback while I commuted. 

BTW just tested this with the hello-samza and it seems to fix the exception. 

Stay tuned!

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java [hunk 2]
```diff
@@ -62,6 +67,30 @@ private static Properties mapToProperties(Map<String, String> map) {
...
+   *
+   * @param originalConfig  The original config to filter
+   * @return                The filtered config
+   */
+  private static Map<String, String> filterUnsupportedProperties(Map<String, String> originalConfig) {
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java]: Can the return type me ImmutableMap instead?

File: samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java [hunk 3]
```diff
@@ -62,6 +67,30 @@ private static Properties mapToProperties(Map<String, String> map) {
...
+      // Kafka requires replication factor, but not as a property, so we have to filter it out.
+      if (!KafkaConfig.TOPIC_REPLICATION_FACTOR().equals(entry.getKey())) {
+        if (LogConfig.configNames().contains(entry.getKey())) {
+          filteredConfig.put(entry.getKey(), entry.getValue());
+        } else {
+          LOG.warn("Property '{}' is not a valid Kafka topic config. It will be ignored.");
+        }
+      }
+    }
+    return filteredConfig;
```
Comment (bharathkk) [inline samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaStreamSpec.java]: Can we java8 'fy this?

**Pull request closed without merging**